### DISCUSSION TITLE
How to include files and directories in consult-fd or consult-find?

### DISCUSSION THREAD

**Discussion by rickswe** [minad/consult#1033] [Q&A]
Hello,

I have a use case in which sometimes I need to include files and directories. However, the arguments only allow
for DIR `&optional DIR`. How can I include both files and directories in both consult-fd and or consult-find?

I would like to search all of my emacs files but I would like to include `early-init.el` and `init.el` along with
everything in my `lisp/` directory.

This will get me an error:

```text
fd error Search path ...early-init.el is not a directory
```

```lisp
(let* ((files-or-dirs (mapconcat
                         (lambda (dir-or-file)
                             (expand-file-name  dir-or-file user-emacs-directory)) '("init.el" "early-init.el" "lisp/") " "))
      (consult-fd-args (format "fd %s" files-or-dirs)))
      (consult-fd))

;; Output
;; fd ~/.config/emacs/init.el ~/.config/emacs/early-init.el ~/.config/emacs/init/
 ```

Thanks

**Comment by minad** [comment]
This cannot be supported if `fd` only supports directories as part of its search path.

**Reply by rickswe** [reply]
Can this be done with find or any other workarounds?
